aiohttp/streams.py

```diff
@@ -663,42 +661,5 @@ async def read(self) -> _SizedT:
             raise self._exception
         raise EofStream
 
-    def __aiter__(self) -> AsyncStreamIterator[_SizedT]:
+    def __aiter__(self) -> AsyncStreamIterator[_T]:
         return AsyncStreamIterator(self.read)
-
-
-class FlowControlDataQueue(DataQueue[_SizedT]):
-    """FlowControlDataQueue resumes and pauses an underlying stream.
-
-    It is a destination for parsed data.
-    """
-
-    def __init__(
-        self, protocol: BaseProtocol, limit: int, *, loop: asyncio.AbstractEventLoop
-    ) -> None:
-        super().__init__(loop=loop)
-        self._size = 0
-        self._protocol = protocol
-        self._limit = limit * 2
-
-    def feed_data(self, data: _SizedT) -> None:
-        self._size += len(data)
-        self._buffer.append(data)
-        if (waiter := self._waiter) is not None:
-            self._waiter = None
-            set_result(waiter, None)
-        if self._size > self._limit and not self._protocol._reading_paused:
-            self._protocol.pause_reading()
```

Comment:
Not sure if this is still relevant? If so, it could just use `len(self._buffer)` now. But, the old code would have tripled the size (i.e. size is incremented by 3 for every message).

Reply:
I guess this is what you're referring to in the PR description. But, my point is that we don't need to track the _size attribute at all for this logic (I assumed this was for when the generic was `bytes`, but don't see anywhere in the code it's actually used like that currently).

Reply:
Alternatively, maybe _size should have been tracking `len(msg.data)` instead, in order to limit the total amount of bytes?

Currently, it's `len(msg)` which is always 3, and clearly not useful.

Reply:
I think your previous comment is proving it's not very useful and supports removing it